jdaviz/app.py

```diff
@@ -91,206 +133,567 @@ def __init__(self, configuration=None, *args, **kwargs):
+        -------
+        `glue.core.Hub`
+            The internal ``Hub`` instance for the application.
+        """
         return self._application_handler.data_collection.hub
 
     @property
     def session(self):
+        """
+        Reference to the stored session instance.
+
+        Returns
+        -------
+        `glue.core.Session`
+            The ``Session`` instance maintained by Glue for this application.
+        """
         return self._application_handler.session
 
     @property
     def data_collection(self):
+        """
+        Reference to the stored data collection instance, used to maintain the
+        the data objects that been loaded into the application this session.
+
+        Returns
+        -------
+        `glue.core.DataCollection`
+            The ``DataCollection`` instance for this application session.
+        """
         return self._application_handler.data_collection
 
-    def vue_on_state_changed(self, *args, **kwargs):
-        print(args, kwargs)
-
     def load_data(self, path):
+        """
+        Provided a path to a data file, open and parse the data into the
+        `~glue.core.DataCollection` for this session. This also attempts to
+        find WCS links that exist between data components.
+
+        Parameters
+        ----------
+        path : str
+            File path for the data file to be loaded.
+        """
         self._application_handler.load_data(path)
 
         # Attempt to link the data
         links = find_possible_links(self.data_collection)
 
-        self.data_collection.add_link(links['Astronomy WCS'])
-
-    def viewers(self, reference):
-        return self._viewer_references.get(reference)
-
-    def get_data(self, reference, cls=None):
-        viewer = self.viewers(reference)
-
-        data = [layer_state.layer
+        # self.data_collection.add_link(links['Astronomy WCS'])
+
+    def get_viewer(self, reference):
+        """
+        Return a `~glue_jupyter.bqplot.common.BqplotBaseView` viewer instance.
+        This is *not* an ``IPyWidget``. This is stored here because
+        the state of the viewer and data methods that allow add/removing data
+        to the viewer exist in a wrapper around the core ``IPyWidget``, which
+        is needed to interact with the data rendered within a viewer.
+
+        Notes
+        -----
+            This is only used in cases where the viewers have been pre-defined
+            in the configuration file. Otherwise, viewers are not stored via
+            reference.
+
+        Parameters
+        ----------
+        reference : str
+            The reference to the viewer defined with the ``reference`` key
+            in the yaml configuration file.
+
+        Returns
+        -------
+        `~glue_jupyter.bqplot.common.BqplotBaseView`
+            The viewer class instance.
+        """
+        return self._viewer_by_reference(reference)
+
+    def get_viewer_data(self, reference, label=None, cls=None):
```

Comment:
I think this name is a bit backward.  What about `get_data_from_viewer`? That feels more like something I'd look for to get the data, as opposed to `get_viewer_data` which sounds like "get some data about the viewer"

Reply:
Done.